src/app/qgsloadstylefromdbdialog.h

```diff
@@ -33,12 +36,17 @@ class APP_EXPORT QgsLoadStyleFromDBDialog : public QDialog, private Ui::QgsLoadS
 
     void initializeLists( const QStringList& ids, const QStringList& names, const QStringList& descriptions, int sectionLimit );
     QString getSelectedStyleId();
+    void selectionChanged( QTableWidget *styleTable );
+
+    void setLayer( QgsVectorLayer *l );
 
   public slots:
-    void cellSelectedRelatedTable( int r );
-    void cellSelectedOthersTable( int r );
+    void relatedTableSelectionChanged();
+    void otherTableSelectionChanged();
+    void deleteStyleFromDB();
 
   private:
+    QgsVectorLayer *mLayer;
```

Comment:
From a quick glance it seems this isn't initialised. Can you change this to:

    QgsVectorLayer* mLayer = nullptr;